packages/typespec-vscode/src/typespec-utils.ts

```diff
@@ -53,3 +53,22 @@ export async function TraverseMainTspFileInWorkspace() {
         .map((uri) => normalizeSlashes(uri.fsPath)),
     );
 }
+
+type TspCliType = "tsp-npm-global" | "tsp-standalone" | "not-available";
+export async function checkTspCliType(): Promise<TspCliType> {
+  try {
+    // use "where" or "which" command instead of "tsp --version" which is much slower
+    const command = process.platform === "win32" ? "where" : "which";
+    const r = await spawnExecution(command, ["tsp"], process.cwd());
+    if (r.exitCode !== 0) {
+      return "not-available";
+    } else {
+      const founds = r.stdout.split("\n").filter((line) => !isWhitespaceStringOrUndefined(line));
+      // the standalone tsp is expected to be installed at .../.tsp/... folder
```

Comment:
this is not an assumption we should make. Even though the install script is doing that, it might not do that in the future and people should be able to install it themself somewhere else

Reply:
then any other way to determine whether the cli is standalone? How about checking the "compiler_installs" at the same folder which I believe user can't customize? Or maybe checking the compiler_installs/package.json if you can put some special thing there (i.e. name = standalone tsp)? I would prefer the first one which can save a fileRead IO.

Reply:
btw, after I run the installation script, I found we are adding ".../.tsp" folder instead of ".../.tsp/bin" folder into PATH env var in linux, which makes ```tsp``` command actually doesn't work because it can't be found through PATH. Shall i file an issue for it? Or did I miss any step on linux?

Reply:
```
tsp --version
```
logs

`standalone 0.65.0` instead of just `0.65.0`

`compiler_installs` is also not a solution this will go as soon as we can, this download the compiler is not a long term solution

Reply:
> btw, after I run the installation script, I found we are adding ".../.tsp" folder instead of ".../.tsp/bin" folder into PATH env var in linux, which makes tsp command actually doesn't work because it can't be found through PATH. Shall i file an issue for it? Or did I miss any step on linux?

yeah that seems like a bug

Reply:
#6237 is filed for the bug. And updated to check ```node``` instead of ```tsp``` so we can still make the call while avoid the perf impact from ```tsp --version```.

Reply:
that's fine for now but this is technically worse as the user could have an old version of node that doesn't work on the path but typespec cli installed